### PR TITLE
Validate mime type in productImage mutations

### LINKED ISSUE #2321: Mutation productImageCreate doesn't validate file extension
The mutation `productImageCreate` which is used to upload product images from the dashboard doesn't validate file extensions. Now it is possible to upload e.g. an audio file.

### PR DESCRIPTION
**Pull request by oldPadavan**
I want to merge this change because...

Fixes #2321

### Pull Request Checklist

1. [x] Privileged views and APIs are guarded by proper permission checks.
1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Database queries are optimized and the number of queries is constant.
1. [x] The changes are tested.
1. [x] The code is documented (docstrings, project documentation).

### CODE REVIEW

File: saleor/graphql/product/mutations/products.py [hunk 1]
```diff
@@ -461,6 +469,14 @@ class Meta:
         description = 'Updates a product image.'
         model = models.ProductImage
 
+    @classmethod
```
Comment (NyanKiyoshi) [inline saleor/graphql/product/mutations/products.py]: You should not write the same code twice.

File: saleor/graphql/product/mutations/products.py [hunk 2]
```diff
@@ -444,6 +444,14 @@ class Meta:
         description = 'Creates a product image.'
         model = models.ProductImage
 
+    @classmethod
+    def clean_input(cls, info, instance, input, errors):
+        cleaned_input = super().clean_input(info, instance, input, errors)
+        uploaded_image = cleaned_input['image']
+        if not uploaded_image.content_type.startswith('image'):
```
Comment (NyanKiyoshi) [inline saleor/graphql/product/mutations/products.py]: The correct mime type is `image/...` (https://www.iana.org/assignments/media-types/media-types.xhtml#image), you should check it's starting with `image/`.

File: saleor/graphql/product/mutations/products.py (hunk 1 above)
Comment (maarcingebala) [reply saleor/graphql/product/mutations/products.py]: True, `ProductImageUpdate` could inherit from `ProductImageCreate`.

**Comment by oldPadavan** [comment]
By the way, I've noticed the test `test_product_image_update_mutation` seems to be actually testing `productImageCreate` mutation instead of `productImageUpdate`
```
def test_product_image_update_mutation(admin_api_client, product_with_image):
    product = product_with_image
    query = """
    mutation updateProductImage($image: Upload!, $alt: String, $product: ID!) {
        productImageCreate(input: {image: $image, alt: $alt, product: $product}) {
            productImage {
                alt
            }
        }
    }
    """
```

**Pull request merged**